server/container/guice/blob-objectstorage-guice/src/test/java/org/apache/james/modules/objectstorage/PayloadCodecProviderTest.java

```diff
@@ -0,0 +1,175 @@
+        FakePropertiesProvider.builder()
+            .register("objectstorage",
+                newConfigBuilder().build()
+            ).build();
+
+    private static final FakePropertiesProvider AES_PROPERTIES_PROVIDER =
+        FakePropertiesProvider.builder()
+            .register("objectstorage",
+                newConfigBuilder()
+                    .put("objectstorage.payload.codec", PayloadCodecs.AES256.name())
+                    .put("objectstorage.aes256.hexsalt", "12345123451234512345")
+                    .put("objectstorage.aes256.password", "james is great")
+                    .build())
+            .build();
+    private static final FakePropertiesProvider MISSING_SALT_PROPERTIES_PROVIDER =
+        FakePropertiesProvider.builder()
+            .register("objectstorage",
+                newConfigBuilder()
+                    .put("objectstorage.payload.codec", PayloadCodecs.AES256.name())
+                    .put("objectstorage.aes256.password", "james is great")
+                    .build())
+            .build();
+    private static final FakePropertiesProvider MISSING_PASSWORD_PROPERTIES_PROVIDER =
+        FakePropertiesProvider.builder()
+            .register("objectstorage",
+                newConfigBuilder()
+                    .put("objectstorage.payload.codec", PayloadCodecs.AES256.name())
+                    .put("objectstorage.aes256.hexsalt", "12345123451234512345")
+                    .build())
+            .build();
+
+    @Test
+    void shouldBuildADefaultPayloadCodecForDefaultConfig() throws Exception {
+        PayloadCodec payloadCodec = new PayloadCodecProvider(DEFAULT_PROPERTIES_PROVIDER).get();
+        assertThat(payloadCodec).isInstanceOf(DefaultPayloadCodec.class);
+    }
+
+    @Test
+    void shouldBuildAnAESPayloadCodecForAESConfig() throws Exception {
+        PayloadCodec payloadCodec = new PayloadCodecProvider(AES_PROPERTIES_PROVIDER).get();
+        assertThat(payloadCodec).isInstanceOf(AESPayloadCodec.class);
+    }
+
+    @Test
+    void shouldFailIfCodeKeyIsMissing() throws Exception {
+        assertThatThrownBy(() -> new PayloadCodecProvider(EMPTY_PROPERTIES_PROVIDER).get()).isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    void shouldFailForAESCodecWhenSaltKeyIsMissing() throws Exception {
+        assertThatThrownBy(() -> new PayloadCodecProvider(MISSING_SALT_PROPERTIES_PROVIDER).get()).isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    void shouldFailForAESCodecWhenPasswordKeyIsMissing() throws Exception {
+
+        assertThatThrownBy(() -> new PayloadCodecProvider(MISSING_PASSWORD_PROPERTIES_PROVIDER).get()).isInstanceOf(IllegalArgumentException.class);
+    }
+
+    private static MapConfigurationBuilder newConfigBuilder() {
+        return new MapConfigurationBuilder();
+    }
+
+    private static class FakePropertiesProvider extends PropertiesProvider {
```

Comment:
Why not make this a standalone class to help tesing configuration in Guice project?